### DISCUSSION TITLE
How to enable "HLK-LD2410"?

### DISCUSSION THREAD

**Discussion by GeorgeQii** [arendst/Tasmota#17909] [Q&A] [answered]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._ 

I browsed "https://tasmota.github.io/docs/Supported-Peripherals/" and I think Tasmota already supports the HLK-LD2410, so I bought one. I've been working on this "peripheral" for the past few hours, but the serial tx/rx doesn't work for me. I even tried the Tuya tx/rx and it didn't work either.

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [ ] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [ ] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [ ] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [ ] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ ] Device used (e.g., Sonoff Basic): _____
- [ ] Tasmota binary firmware version number used: _____
  - [ ] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: _____
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by sfromis** [comment, marked as answer]
You need a custom build to include this optional feature, `USE_LD2410`.

(And you should not open an "issue" to ask questions, using "Discussions" is better for that, or on the Tasmota Discord server.

**Reply by liv-in-sky** [reply]
hi , could you please help me - I'm not very expierenced with tasmota  - I have tried to make a bin file with Gitpod. where do I have to integrate USE_LD2410. I did it in the "my_user_config.h" file. My wifi credentials (I added these) are working after flashing the file. but I'm not able to choose the sensor . Where do I have to insert the USE_LD2410 ? Do I have to load an additional definition file or something like this?

thanx in advance

**Reply by sfromis** [reply]
The recommendation is that you do not make any changes to `my_user_config.h`, but do all your changes in `user_config_override.h`, making sure to not do it within the big comment block with examples of how to use the file.

Read https://tasmota.github.io/docs/Compile-your-build/#customize-your-build

**Comment by palmerfarmer50** [comment]
Ok so similar to this, I am hacking one of those VINDRIKTNING units from IKEA with a D1 wemos and need to add all the sensors that arent include in the basic tasmota bin file (not all are in the 'allsensors' version either) so these were added seperately, at the end on the user_config_overide.h file
BME280 (temp hum baro), LD2410 (presence sensor - not tested because its still in the post), BH1750 (luminance level), PM1006K (VINDRIKTNING dust sensor)
There is a really good and quick video on how to do this on, 
https://www.youtube.com/watch?v=R6KeZF6ktLc

this is at the bottom of the user_config_overide.h file
(someone will probably correct me about the end if statements but i'm not a programmer more of a hacker of other peoples work!)


#define USE_VINDRIKTNING    // Add support for IKEA VINDRIKTNING particle concentration sensor (+1k code)¸

#ifndef USE_BMP
#define USE_BMP     // [I2cDriver10] Enable BMP085/BMP180/BMP280/BME280 sensors (I2C addresses 0x76 and 0x77) (+4k4 code)
#endif

#ifndef USE_BH1750
#define USE_BH1750      // [I2cDriver11] Enable BH1750 sensor (I2C address 0x23 or 0x5C) (+0k5 code)
#endif

#ifndef USE_LD2410     // Human Presence detector
#define USE_LD2410
#endif

**Reply by barbudor** [reply]
Your lines are correct but the location you placed the is key
Don't insert it in the greyed areas
The 3 green lines gives valid locations

Actually you can remove all greyed lines between the #define and the #endif


![IMG_20240127_172856](https://github.com/arendst/Tasmota/assets/2996042/5b7f4be2-3bd8-479e-a5f5-96bc7300c022)

**Reply by sfromis** [reply]
When you do something in the block with examples, you're just adding to the comments there, not used for anything but human reading. Of course, it may be helpful to have read the note at the end of the comment block :wink:
